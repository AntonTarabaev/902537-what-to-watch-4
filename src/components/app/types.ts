import {Comment, Film} from "@root/types";
import {AppActionTypes} from "@constants/action-types";

export interface DataState {
  films: Film[];
  promo: Film;
  comments: Comment[];
  isLoaded: boolean;
}

interface SetFilmsAction {
  type: typeof AppActionTypes.SET_FILMS;
  payload: Film[];
}

interface SetPromoAction {
  type: typeof AppActionTypes.SET_PROMO;
  payload: Film;
}

interface SetCommentsAction {
  type: typeof AppActionTypes.SET_COMMENTS;
  payload: Comment[];
}

interface SetLoadedStatusAction {
  type: typeof AppActionTypes.SET_LOADED_STATUS;
  payload: boolean;
}

interface SetFilmIsFavoriteAction {
  type: typeof AppActionTypes.SET_FILM_IS_FAVORITE;
  payload: Film;
}

export type DataActions = SetCommentsAction | SetFilmIsFavoriteAction | SetFilmsAction | SetPromoAction | SetLoadedStatusAction;
