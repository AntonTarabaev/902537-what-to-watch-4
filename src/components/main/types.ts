import {GenresListActionTypes, MainPageActionTypes} from "@constants/action-types";

export interface MainPageState {
  filterGenre: string;
  showingFilmsCount: number;
}

interface IncrementShowingFilmsAction {
  type: typeof MainPageActionTypes.INCREMENT_SHOWING_FILMS;
  payload: number;
}

interface ChangeActiveGenreAction {
  type: typeof GenresListActionTypes.CHANGE_GENRE_FILTER;
  payload: string;
}

interface ResetShowingFilmsAction {
  type: typeof GenresListActionTypes.RESET_SHOWING_FILMS;
  payload: number;
}

export type MainPageActions = IncrementShowingFilmsAction | ChangeActiveGenreAction | ResetShowingFilmsAction;
