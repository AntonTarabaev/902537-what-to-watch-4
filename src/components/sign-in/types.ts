import {AuthorizationStatus, Film, User} from "@root/types";
import {UserActionTypes} from "@constants/action-types";

export interface UserState {
  authorizationStatus: AuthorizationStatus;
  userData: User;
  favoriteFilms: Film[];
}

interface RequireAuthorizationAction {
  type: typeof UserActionTypes.REQUIRED_AUTHORIZATION;
  payload: AuthorizationStatus;
}

interface SetUserDataAction {
  type: typeof UserActionTypes.SET_USER_DATA;
  payload: User;
}

interface SetFavoriteAction {
  type: typeof UserActionTypes.SET_FAVORITE;
  payload: Film[];
}

export type UserActions = RequireAuthorizationAction | SetUserDataAction | SetFavoriteAction;
