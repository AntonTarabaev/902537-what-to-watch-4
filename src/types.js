export const Film = PropTypes.shape({
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired,
  bgImage: PropTypes.string.isRequired,
  previewImage: PropTypes.string.isRequired,
  releaseYear: PropTypes.number.isRequired,
  genre: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  ratingVotes: PropTypes.number.isRequired,
  director: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  actors: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  preview: PropTypes.string.isRequired,
  duration: PropTypes.number.isRequired,
  video: PropTypes.string.isRequired,
  isFavorite: PropTypes.bool.isRequired,
}).isRequired;

export const Comment = PropTypes.shape({
  author: PropTypes.string.isRequired,
  comment: PropTypes.string.isRequired,
  date: PropTypes.instanceOf(Date).isRequired,
  rating: PropTypes.number.isRequired,
}).isRequired;
