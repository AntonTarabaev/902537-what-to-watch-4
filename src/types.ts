export enum AuthorizationStatus {
  AUTH = `AUTH`,
  NO_AUTH = `NO_AUTH`,
}

export interface Film {
  id: string;
  title: string;
  poster: string;
  bgImage: string;
  previewImage: string;
  releaseYear: number;
  genre: string;
  rating: number;
  ratingVotes: number;
  director: string;
  description: string;
  actors: string[];
  preview: string;
  duration: number;
  video: string;
  isFavorite: boolean;
}

export interface Comment {
  id: number;
  author: string;
  authorId: number;
  comment: string;
  date: Date;
  rating: number;
}
