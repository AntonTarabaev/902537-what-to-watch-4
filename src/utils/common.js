import {MINUTES_IN_HOUR, MONTHS} from "@constants/main";

export const formatTime = (durationInMinutes) => {
  return (
    durationInMinutes > MINUTES_IN_HOUR ?
      `${(durationInMinutes - durationInMinutes % MINUTES_IN_HOUR) / MINUTES_IN_HOUR}h ${durationInMinutes % MINUTES_IN_HOUR}m` :
      `${durationInMinutes}m`
  );
};

export const formatDate = (date) => {
  return `${MONTHS[date.getMonth()]} ${date.getDate()}, ${date.getFullYear()}`;
};
