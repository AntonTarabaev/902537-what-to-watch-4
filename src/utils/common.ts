import {MINUTES_IN_HOUR, MONTHS} from "@constants/main";
import {Film} from "@root/types";

export const formatTime = (durationInMinutes: number) => {
  return (
    durationInMinutes > MINUTES_IN_HOUR ?
      `${(durationInMinutes - durationInMinutes % MINUTES_IN_HOUR) / MINUTES_IN_HOUR}h ${durationInMinutes % MINUTES_IN_HOUR}m` :
      `${durationInMinutes}m`
  );
};

export const formatDate = (date: Date) => {
  return `${MONTHS[date.getMonth()]} ${date.getDate()}, ${date.getFullYear()}`;
};

export const extend = (a: object, b: object) => {
  return Object.assign({}, a, b);
};

export const getUniqueGenres = (films: Film[], maxGenresCount: number) => {
  return [...new Set(films.map((it) => it.genre))].slice(0, maxGenresCount);
};

export const changeFormElementsDisabledProperty = (form: HTMLFormElement, disabled: boolean) => {
  Array.from(form.elements).map((formElement: HTMLInputElement) => {
    formElement.disabled = disabled;
  });
};

export const noop = () => {
  // do nothing
};
