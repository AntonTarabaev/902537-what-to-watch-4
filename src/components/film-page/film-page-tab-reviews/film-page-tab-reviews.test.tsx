import FilmPageTabReviews from "@components/film-page/film-page-tab-reviews/film-page-tab-reviews";
import {Comment} from "@root/types";
import * as renderer from "react-test-renderer";
import {noop} from "@utils/common";

const comments: Comment[] = [
  {
    id: 1,
    author: `John Doe`,
    authorId: 123,
    rating: 8.9,
    date: new Date(8475893),
    comment: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Sed sed nisi sed augue convallis suscipit in sed felis. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
  },
  {
    id: 2,
    author: `Robert Rodrigues`,
    authorId: 321,
    rating: 5.4,
    date: new Date(4231284),
    comment: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna`,
  }
];

it(`Should ReviewsList render correctly`, () => {
  const tree = renderer.create(
      <FilmPageTabReviews
        filmId={`123`}
        loadFilmComments={noop}
        setFilmComments={noop}
        comments={comments}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
