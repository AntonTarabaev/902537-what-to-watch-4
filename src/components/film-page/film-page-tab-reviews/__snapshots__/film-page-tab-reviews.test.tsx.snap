// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ReviewsList render correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Sed sed nisi sed augue convallis suscipit in sed felis. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            John Doe
          </cite>
          <time
            className="review__date"
            dateTime="1970-01-01"
          >
            January 1, 1970
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8.9
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Robert Rodrigues
          </cite>
          <time
            className="review__date"
            dateTime="1970-01-01"
          >
            January 1, 1970
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        5.4
      </div>
    </div>
  </div>
</div>
`;
